### PR TITLE
Edit target

### PR DESCRIPTION
**Pull request by mato2593**
This PR includes the edition of a previously created target.

https://trello.com/c/WBciALye/17-3-as-a-user-i-should-be-able-to-edit-a-previously-created-target

edit-target

### CODE REVIEW

**Review by mjurfest — APPROVED** [review]

File: Target-MVD-Mathias/Networking/Services/TargetAPI.swift [hunk 1]
```diff
@@ -81,4 +81,32 @@ class TargetAPI {
...
+    ] as [String : Any]
+    
+    APIClient.sendPutRequest(url, params: params as [String : AnyObject], success: { (response) in
+      let json = JSON(response)
+      let target = Target.parse(fromJSON: json["target"])
+      
+      let compatibleTargetsArray = json["matches"].arrayValue
+      var compatibleTargets: [Target] = []
+      
+      for compatibleTarget in compatibleTargetsArray {
```
Comment (agch-dev) [inline Target-MVD-Mathias/Networking/Services/TargetAPI.swift]: Parsing a lot of Targets at once seems like something you will have to do more than once, you should consider moving this to the Target model and make a general Target array parser.

File: Target-MVD-Mathias/Views/TargetFormView.swift [hunk 2]
```diff
@@ -204,15 +241,28 @@ class TargetFormView: UIView {
...
+  private func targetEdited() -> Bool {
+    var targetEdited = false
+    
+    if let target = target {
+      targetEdited = target.title != title || target.radius != area || target.topic != topic
+    }
+    
+    return targetEdited
+  }
+
```
Comment (agch-dev) [inline Target-MVD-Mathias/Views/TargetFormView.swift]: Remove extra line.

File: Target-MVD-Mathias/ViewControllers/HomeViewController.swift [hunk 3]
```diff
@@ -302,6 +347,33 @@ extension HomeViewController: UITableViewDelegate {
                       animations: {
                         self.topicsTableView.center.y += self.topicsTableView.frame.size.height
     })
+    
+    showingTopicsTableView = false
+  }
+
```
Comment (agch-dev) [inline Target-MVD-Mathias/ViewControllers/HomeViewController.swift]: Remove extra line

File: Target-MVD-Mathias/Networking/Services/TargetAPI.swift [hunk 4]
```diff
@@ -81,4 +81,32 @@ class TargetAPI {
...
+        compatibleTargets.append(Target.parse(fromJSON: compatibleTarget))
+      }
+      
+      success(target, compatibleTargets)
+    }) { (error) in
+      failure(error)
+    }
+  }
+
```
Comment (agch-dev) [inline Target-MVD-Mathias/Networking/Services/TargetAPI.swift]: Remove extra line

**Review by agch-dev — CHANGES_REQUESTED** [review]
Great work! Just a few comments.

**Review by pMalvasio — APPROVED** [review]

File: Target-MVD-Mathias/Networking/Services/TargetAPI.swift (hunk 1 above)
Comment (mato2593) [reply Target-MVD-Mathias/Networking/Services/TargetAPI.swift]: Done!

**Review by agch-dev — APPROVED** [review]

**Pull request merged**